guide_drafts/model-derives.md

```diff
@@ -0,0 +1,769 @@
+
+There are some considerations to be made when implementing the `Queryable` trait. You must include all the table's fields in your model struct. The implementation details for this rule are only important if you wish to `impl` the trait by hand and will be covered later in this section.  Your struct name should be the singular of its corresponding table name. Queryable by itself does not support the `#[table_name=""]` attribute. 
+
+### Example 
+
+*Note: src/lib.rs with #[macro_use] extern crate diesel_codegen; not shown*
+
+*src/models.rs*
+```rust
+#[derive(Queryable)]
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: String,
+}
+```
+
+*src/main.rs*
+```rust
+// externs, database connection, and using statement code omitted
+
+fn main() {
+    // The following will return all Users as a QueryResult<Vec<User>>
+    let query_result = users.load::<User>(&db_connection)
+      .expect("Error loading users");
+}
+```
+
+When deriving Queryable for a model, the struct must have all fields defined. Queryable's associated type, Row,  implements `FromSqlRow`, which provides the `fields_needed()` method. That method returns the number of fields to be consumed by the row. If we were to comment out all the three `String` fields on our User struct, we would see the following error.
+
+```rust
+error[E0277]: the trait bound `(i32,): diesel::types::FromSqlRow<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` is not satisfied
+  --> src/bin/main.rs:21:28
+   |
+21 |     let query_result = users.load::<User>(&db_connection).expect("Error loading users");
+   |                            ^^^^ the trait `diesel::types::FromSqlRow<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` is not implemented for `(i32,)`
+   |
+   = help: the following implementations were found:
+             <(A,) as diesel::types::FromSqlRow<(SA,), DB>>
+   = note: required because of the requirements on the impl of `diesel::Queryable<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` for `diesel_demo_cli::models::User`
+   = note: required because of the requirements on the impl of `diesel::LoadQuery<_, diesel_demo_cli::models::User>` for `diesel_demo_cli::schema::users::table`
+```
+
+Notice the compiler is indicating a trait is not implmented for using the `.load()` method. FromSqlRow is expecting a tuple that looks like `(i32, std::string::String, std::string::String, std::string::String)`, but we currently only have a tuple consisting of `(i32,)`.
```

Comment:
Do you want to talk a bit more about how to parse that error message? I don't think it's necessary, but I talked a bit about that [here](https://deterministic.space/hook-into-rustc-errors.html), feel free to adapt/copy it.